### ISSUE TITLE
Mason Integration Issues

### ISSUE THREAD

**Issue by daniel-j-h** [Project-OSRM/osrm-backend#3234]
The following tests were made on completely fresh environments, e.g.

    docker run -it ubuntu:16.04 /bin/bash

##### Ubuntu 16.04

I expected these packages to be required for mason:

    apt-get install git cmake curl

but it seems like I also need

    libstdc++-5-dev
    pkg-config

with those, I get

```
CMake Error at third_party/mason/mason.cmake:187 (message):
  [Mason] Package at mason_packages/linux-x86_64/expat/2.1.1 has version '',
  but required '2.1.1'
```

##### Ubuntu 15.04

zlib seems to be missing as dependency

```
CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
```

with zlib from apt, the stdlib is too old so `libstdc++5-dev` is needed, too.

**Comment by springmeyer** [comment]
Working on fixes in #3235. The problems were:

 - The `expat` package had an empty `mason.ini` so metadata could not be correctly queried. This is fixed by upgrading to a new `expat` package. Tracking the underlying issue at https://github.com/mapbox/mason/issues/274.

 - `pkg-config` is no longer required for mason integration after the `expat` fix. However it is required of course for the build of the example. But that has always been that way. Now `pkg-config` is no longer a prereq for `ENABLE_MASON`.

Other commenets:

 - Yes, `libstdc++-5-dev` is a requirement. We can loosen this if desirable by rebuilding boost, tbb, and luabind against an older libstdc++. Under discussion at https://github.com/mapbox/mason/issues/267.

 - `zlib` is depended upon via `find_package` in the same way as non mason builds. So this is by design right now and to ensure we link against the system zlib. This is precautionary and forward looking to allow the ENABLE_MASON built `libosrm.a` to be linked to the the node.js binaries provided by https://nodejs.org/en/download/current/ which also link system zlib. Said another way: It would be easy to link to a mason-provided zlib.a, but this might not work for node-osrm once we integrate it into osrm-backend. And most systems have zlib installed already, so this is a minor annoyance. On debian systems you'll need `apt-get install zlib1g-dev`.

I will address the libstdc++ and zlib needs in the docs: #3207

**Comment by springmeyer** [comment]
Closing, solved by #3235
